platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,493 @@
+   file. Services and DevOps will implement this.
+
+2. Off-by-1 problem: Node IDs are 1 less than the node name. For example, the name of the node with node id 0
+   is `node1`. This is confusing. The node name is used as the alias in the cryptography and used to name the pem files
+   on disk. Node id 0 gets its cryptography through “node1” alias.
+   The resolution is to get rid of node names and use Node IDs only. Services and DevOps will implement this.
+
+3. Inversion of Control: The `config.txt` file does not have all the information that the Hedera Address Book needs (
+   proxy endpoints, and in the future, Block Nodes). It has verbose information for the platform. Its format could be
+   better, and it stores the account number in the memo field. Upon the creation and adoption of rosters in the
+   state, `config.txt` is no longer useful.
+   The resolution is to offload this duty off the platform and allow Services to use whatever file format that suits
+   them as described earlier. Services will implement this.
+
+### Roster changes needed for Components
+
+#### Reconnect
+
+Reconnect. Reconnect logic currently exchanges and validates the address book between the learner and teacher nodes. The
+learner node uses the address book to select the teacher node, as well as remove all invalid signatures from the
+state. Both of these will need to be updated to use rosters instead.
+
+#### Others
+
+- Networking. Most parts of the network code have been abstracted away from the address book as part of the Dynamic
+  Address Book effort, so there should be minimal work left there.
+- Event Validation. The event validation logic uses the address book to determine whether a given event has a valid
+  signature. This will be updated to use rosters instead.
+- Miscellaneous components. Other components that reference things stored in the address book (like consensus needing
+  the node weight or the crypto module verifying keys) will need to be updated to use things stored in rosters instead.
+  Some of these components include SelfEventSigner, EventHasher, HealthMonitor, SignedStateSentine, PcesSequencer,
+  StateHasher, HashLogger etc.
+
+## Test Plan
+
+Some of the obvious test cases to be covered in the plan include validating one or more of the following scenarios:
+
+1. New valid Candidate Roster created with no subsequent one sent by App. Verify accept.
+2. New valid Candidate Roster created with the subsequent one sent by App. Verify accept.
+3. Invalid roster(s) sent by the App. Verify reject.
+4. Node Failures During Roster Change: What happens if nodes fail or disconnect during a roster change? Verify valid
+   node successfully reconnects.
+5. Concurrent Roster Updates: What if we make multiple roster updates concurrently? Verify no effect on adoption.
+6. Roster recovery? The Node receives a candidate roster, crashes. Wake up, reconnect. Verify recovery.
```

Comment:
This is the same as #4 so it can be removed

Reply:
they are slightly different test cases. 

#4, the candidate roster didn't get set in the state. But when the node comes back up, if it's in the candidate roster the rest of the network has (now) adopted, it will connect and carry on just fine. 

#6 the candidate roster got set in the state for that node, and does the same test as #4. 

Thinking about it more now, In both, two use cases can be tested; one will be that the nodeId is in the roster, the other will be that it's not. But i think there will probably be more test cases that will be added during implementation.